### ISSUE TITLE
How can I make local histories priority over global histories?

### ISSUE THREAD

**Issue by qwIvan** [jimhester/per-directory-history#19]
I want to using local history and global history in the same time, but the local histories are priority over global histories, so I don't need to toggle.

**Comment by CyberShadow** [comment]
I don't think this is easily possible without also replacing ZSH's history search features. Though ZSH can track multiple history lists, I think only one can be active for recall / search at a time.

**Comment by tmpm697** [comment]
@CyberShadow I'm thinking the same as OP, `per-directory-history` only can be useful if I can use local history and global history in manner that default to global history and sometime we need to narrow search through history so I can enable local history for better processing.

Can you do that without re-compile zsh sourcecode? Currently I don't find this plugin usefull because to use local history I have to enable it as default, but I want to use global history as default but need local history avail in case I want to switch to it.

**Comment by jimhester** [comment]
I think in order to do this you would have to replace zsh's behavior and it is outside the scope of this project.
